tests/lib/bin/version-tool

```diff
@@ -0,0 +1,323 @@
+
+    -eq -ne -gt -ge -lt -le
+
+Version comparison is performed using the selected algorithm.
+
+naive:
+    The algorithm considers each version to be a tuple of integers. Non,
+    integer elements or element fragments are discarded.
+
+    Comparison is performed on by considering the leftmost element in each
+    tuple. First pair of numbers that are not equal determine the result of the
+    comparison. Tuples have unequal length then missing elements are
+    substituted with zero.
+    """,
+        formatter_class=RawTextHelpFormatter,
+    )
+    parser.register("action", "unit-test", _UnitTestAction)
+    parser.add_argument("-v", "--version", action="version", version="1.0")
+    parser.add_argument(
+        "--verbose", action="store_true", help="describe comparison process"
+    )
+
+    parser.add_argument("version_a", metavar="VERSION-A")
+    parser.add_argument("version_b", metavar="VERSION-B")
+
+    # algorithm selection
+    alg_grp = parser.add_mutually_exclusive_group(required=True)
+    alg_grp.add_argument(
+        "--naive",
+        dest="algorithm",
+        action="store_const",
+        const="naive",
+        help="select the naive version comparison",
+    )
+    # relation selection
+    rel_op_grp = parser.add_mutually_exclusive_group(required=True)
+    rel_op_grp.add_argument(
+        "-eq",
+        action="store_const",
+        const="==",
+        dest="rel_op",
+        help="test that versions are equal",
+    )
+    rel_op_grp.add_argument(
+        "-ne",
+        action="store_const",
+        const="!=",
+        dest="rel_op",
+        help="test that versions are not equal",
+    )
+    rel_op_grp.add_argument(
+        "-gt",
+        action="store_const",
+        const=">",
+        dest="rel_op",
+        help="test that version-a is greater than version-b",
+    )
+    rel_op_grp.add_argument(
+        "-ge",
+        action="store_const",
+        const=">=",
+        dest="rel_op",
+        help="test that version-a is greater than or equal to version-b",
+    )
+    rel_op_grp.add_argument(
+        "-lt",
+        action="store_const",
+        const="<=",
+        dest="rel_op",
+        help="test that version-a is less than version-b",
+    )
+    rel_op_grp.add_argument(
+        "-le",
+        action="store_const",
+        const="<=",
+        dest="rel_op",
+        help="test that version-a is less than or equal to version-b",
+    )
+
+    # maintenance commands
+    maint_grp = parser.add_argument_group("maintenance commands")
+    maint_grp.add_argument("--run-unit-tests", action="unit-test", help=SUPPRESS)
+    return parser
+
+
+def main():
+    # type: () -> None
+    parser = _make_parser()
+    opts = parser.parse_args()
+    if opts.algorithm == "naive":
+        delta = naive_version_cmp(opts.version_a, opts.version_b)
+    else:
+        parser.error("unsupported delta algorithm")
+    is_consistent = consistent_relation(opts.rel_op, delta)
```

Comment:
maybe this is just because I haven't read lots of python in a while, but it is a bit odd to declare `delta` within the `if` above, but then use it outside of that branch, no?

Reply:
I don't know, it seems natural to me, this is how python scope works and how I would continue to write code unless I could use Python 3.6+ variable declarations.

Reply:
sure it's fine as is I'll trust your judgment